cmd/check.go

```diff
@@ -59,100 +58,111 @@ func (p *checkParams) regoVersion() ast.RegoVersion {
-	// if capabilities are not provided as a cmd flag,
-	// then ast.CapabilitiesForThisVersion must be called
-	// within checkModules to ensure custom builtins are properly captured
-	if params.capabilities.C != nil {
-		capabilities = params.capabilities.C
-	} else {
+	// ensure custom builtins are properly captured
+	capabilities := params.capabilities.C
+	if capabilities == nil {
 		capabilities = ast.CapabilitiesForThisVersion(ast.CapabilitiesRegoVersion(params.regoVersion()))
 	}
 
+	l := loader.NewFileLoader().
+		WithRegoVersion(params.regoVersion()).
+		WithProcessAnnotation(true).
+		WithCapabilities(capabilities)
+
 	ss, err := loader.Schemas(params.schema.path)
 	if err != nil {
 		return err
 	}
 
+	compiler := ast.NewCompiler().
+		SetErrorLimit(params.errLimit).
+		WithCapabilities(capabilities).
+		WithSchemas(ss).
+		WithEnablePrintStatements(true).
+		WithStrict(params.strict).
+		WithUseTypeCheckAnnotations(true)
+
+	var modules map[string]*ast.Module
+
 	if params.bundleMode {
+		bundles := make([]*bundle.Bundle, 0, len(args))
 		for _, path := range args {
-			b, err := loader.NewFileLoader().
-				WithRegoVersion(params.regoVersion()).
-				WithSkipBundleVerification(true).
-				WithProcessAnnotation(true).
-				WithCapabilities(capabilities).
-				WithFilter(filterFromPaths(params.ignore)).
-				AsBundle(path)
+			b, err := l.WithSkipBundleVerification(true).WithFilter(filterFromPaths(params.ignore)).AsBundle(path)
 			if err != nil {
 				return err
 			}
-			maps.Copy(modules, b.ParsedModules(path))
+			bundles = append(bundles, b)
 		}
-	} else {
-		f := loaderFilter{
-			Ignore:   params.ignore,
-			OnlyRego: true,
+		b, err := bundle.Merge(bundles)
+		if err != nil {
+			return err
 		}
 
-		result, err := loader.NewFileLoader().
-			WithRegoVersion(params.regoVersion()).
-			WithProcessAnnotation(true).
-			WithCapabilities(capabilities).
-			Filtered(args, f.Apply)
+		modules = maps.Clone(b.ParsedModules(""))
+		if len(b.Data) > 0 {
+			compiler = compiler.WithPathConflictsCheck(mapFinder(b.Data))
+		}
+	} else {
+		result, err := l.Filtered(args, ignoredOnlyRego(params.ignore).Apply)
 		if err != nil {
 			return err
 		}
 
-		for _, m := range result.Modules {
-			modules[m.Name] = m.Parsed
-		}
+		modules = result.ParsedModules()
 	}
 
-	compiler := ast.NewCompiler().
-		SetErrorLimit(params.errLimit).
-		WithCapabilities(capabilities).
-		WithSchemas(ss).
-		WithEnablePrintStatements(true).
-		WithStrict(params.strict).
-		WithUseTypeCheckAnnotations(true)
-
-	compiler.Compile(modules)
-	if compiler.Failed() {
+	if compiler.Compile(modules); compiler.Failed() {
 		return compiler.Errors
 	}
+
 	return nil
 }
 
+// emulate storage.NonEmpty without having to create storage / transaction
+// returned function returns false, nil when m or path is empty
+func mapFinder(m map[string]any) func(path []string) (bool, error) {
+	if len(m) == 0 {
+		return func(path []string) (bool, error) {
```

Comment:
I doubt that's it's of consequence here, but if this were a hot path, we might better create a 
```go
func empty([]string) (bool, error) { return false, nil }
```

and reference that here? I don't know if the compiler is clever enough to do that on its own...

Reply:
Fixed 👍